tests/e2e/restart_test.go

```diff
@@ -20,97 +18,130 @@
 package e2e
 
 import (
-	"log"
-	"os"
-
 	. "gopkg.in/check.v1"
 
 	"github.com/vmware/docker-volume-vsphere/tests/utils/dockercli"
 	"github.com/vmware/docker-volume-vsphere/tests/utils/inputparams"
 	"github.com/vmware/docker-volume-vsphere/tests/utils/verification"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/misc"
 )
 
-type PluginSuite struct {
+type RestartTestData struct {
+	config        *inputparams.TestConfig
 	volumeName    string
-	hostIP        string
 	containerName string
-	esxIP         string
 }
 
-func (s *PluginSuite) SetUpTest(c *C) {
-	s.hostIP = os.Getenv("VM2")
-	s.esxIP = os.Getenv("ESX")
+func (s *RestartTestData) SetUpSuite(c *C) {
+	s.config = inputparams.GetTestConfig()
+	if s.config == nil {
+		c.Skip("Unable to retrieve test config, skipping vmgroup tests")
+	}
+
 	s.volumeName = inputparams.GetUniqueVolumeName("restart_test")
 	s.containerName = inputparams.GetUniqueContainerName("restart_test")
 
+	// ensure there are no remanants from an earlier run
+	out, err := dockercli.RemoveContainer(s.config.DockerHosts[1], s.containerName)
```

Comment:
Thats what was in the original code, but the stop container doesn't remove the container so creating a new one fails with "there is a container with the same name already" and test fails. Since kill is being done for  the docker and plugin the containers get stopped not removed.